### ISSUE TITLE
After resuming traing `scheduler.step()` will not update optimzer's learning rate

### ISSUE THREAD

**Issue by zxydi1992** [Lightning-AI/lightning#17539]
### Bug description

Not able to update learning rate after resuming is a critical bug, and make training longer jobs with Pytorch Lightning very hard. 

https://github.com/Lightning-AI/lightning/issues/12812

### What version are you seeing the problem on?

v1_9

### How to reproduce the bug

```python
NA
```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by Borda** [comment]
as here we do not have any more details, let's close it and re-classify the #12812 as bug
